Adaptors/RabbitMQ/src/PushQueueStorage.cs

```diff
@@ -80,29 +83,32 @@ private void PushMessages(IEnumerable<MessageData> messages,
     var queueArgs = new Dictionary<string, object>
                     {
                       {
-                        "x-max-priority", Options!.MaxPriority
+                        "x-max-priority", Options.MaxPriority
                       },
                       {
                         "x-queue-mode", "lazy" // queue will try to move messages to disk as early as practically possible
                       },
                     };
 
-    ConnectionRabbit.Channel!.QueueDeclare(partitionId,
-                                           false, /* to survive broker restart */
-                                           false, /* used by multiple connections */
-                                           false, /* not deleted when last consumer unsubscribes (if it has had one) */
-                                           queueArgs);
+    (await ConnectionRabbit.GetConnectionAsync(cancellationToken)
+                           .ConfigureAwait(false)).QueueDeclare(partitionId,
+                                                                false, /* to survive broker restart */
+                                                                false, /* used by multiple connections */
+                                                                false, /* not deleted when last consumer unsubscribes (if it has had one) */
+                                                                queueArgs);
 
     foreach (var msg in messages)
     {
-      var basicProperties = ConnectionRabbit.Channel!.CreateBasicProperties();
+      var basicProperties = (await ConnectionRabbit.GetConnectionAsync(cancellationToken)
+                                                   .ConfigureAwait(false)).CreateBasicProperties();
```

Comment:
I would also split this into 2 statements, and reuse the connection for the basicPublish just after. But I think it is a good thing to re-call this GetConnectionAsync for every elements.